storm-core/src/jvm/org/apache/storm/cluster/PaceMakerStateStorage.java

```diff
@@ -159,10 +162,6 @@ public void set_worker_hb(String path, byte[] data, List<ACL> acls) {
                         ret = details;
                     }
                 }
-                if(ret == null) {
-                    throw new HBExecutionException("Failed to get a response.");
-                }
-                LOG.debug("Successful get_worker_hb");
                 return ret;
```

Comment:
Do we need the ret == null check before this?  Is it possible to get more then one response and all of them are bogus?